source/lib.ts

```diff
@@ -324,11 +326,24 @@ const rateLimit = (
 
 			// Create an augmented request
 			const augmentedRequest = request as AugmentedRequest
-
 			// Get a unique key for the client
 			const key = await config.keyGenerator(request, response)
 			// Increment the client's hit counter by one.
-			const { totalHits, resetTime } = await config.store.increment(key)
+
+			let totalHits = 0
+			let resetTime
+			try {
+				const incrementResult = await config.store.increment(key)
+				totalHits = incrementResult.totalHits
+				resetTime = incrementResult.resetTime
+			} catch (error) {
+				if (config.passOnStoreError) {
+					next()
```

Comment:
```suggestion
					console.error('express-rate-limit: error from store, allowing request without rate-limiting.', error)
					next()
```

This is what I meant by point 3.